src/page/page_cache.rs

```diff
@@ -109,8 +123,9 @@ impl<M> PageCache<LockFreeLog, M>
                         let stack = raw(Stack::default());
                         self.inner.insert(prepend.pid, stack).unwrap();
                         free_pids.retain(|&pid| pid != prepend.pid);
-                        if self.max_id.load(SeqCst) < prepend.pid {
-                            self.max_id.store(prepend.pid, SeqCst);
+                        if self.max_pid.load(SeqCst) <= prepend.pid {
+                            // always make sure self.max_pid is higher than what exists
+                            self.max_pid.store(prepend.pid + 1, SeqCst);
```

Comment:
add 1 to what we store because self.max_pid should always be higher than what is already present (this turned into a bug when I was writing this)